content/rest-api/beta.textile

```diff
@@ -0,0 +1,165 @@
+    - Batch Publish#batch
+---
+
+h2(#overview). REST API Beta Features
+
+The contents of this section are beta features of our REST API. Although many of these features are fully functional, they are liable to change.
+
+h2(#features). Features
+
+h3(#batch). Batch publish
+
+Batch mode is a way of querying an API multiple times with a single HTTP request. The simplest example to consider is publishing a single message to multiple channels; a single request is made, indicating the channels and the contents of the message, and the system attempts to publish on all channels in parallel. Each of those attempts may succeed or fail independently. The batch mode API provides a way for the results of that request to be indicated back to the caller.
+
+h4(#batch-operation). Generic operation
+
+A batch mode request translates a single set of request details (ie the request body, params and headers) in to an array of requests for the underlying API. Each of those requests is then performed in parallel.
+
+Once each request has completed, the batch-mode response is formulated to be returned to the caller. There are three possible outcomes:
+* all requests succeeded. This is of course the expected case. The response is then an array containing the responses to the constituent queries in request order.
+* the batch request failed prior to attempting the constituent API calls. This could typically be an authorisation failure, an invalid request, or some required element of the service being unavailable. The response is then an error response, with the applicable status code and error response body (with `message@, @code@ and any other relevant error information.
+* one or more constituent API calls failed. In this case the response contains a generic @400@ status code and the response body contains an error object with a specific error code @40020@ signifying an error in a batch operation. The error body also then contains a @batchResponse@ property which in an array, in request order, containing the outcome of each of the constituent API calls.
+This means that caller can usually make the call and have access to the success result or an error in the case of failure. In the case of a partial failure, many callers may just wish to know whether or not the call succeeded fully, in which case they do not need to look at the @batchResponse@ detail. However, callers that wish to know the detail of the outcome of each call can discover this from the @batchResponse@.
+
+h4(#bulk-publish). Bulk publish
+
+The REST API has been extended to include batch-mode publish operations. The API concerned is:
+
+h6. POST /messages
+
+
+This is the new bulk publish API which is implicitly a batch-mode API. This performs a publish in parallel of messages to channels specified in the request body. Check out "the code example":https://jsbin.ably.io/okehuf/1/edit on how to do this with both the REST API and the "Request":/rest/usage#request method in the REST client library.
```

Comment:
Request should be `@request@` as it's a code definition

Reply:
You should not link directly to a JSbin, we have a helper to look up JSBins automatically and link to them, so that if you change the JSBin in future, it will automatically be reflected here.